modules/dnn/test/test_model.cpp

```diff
@@ -268,6 +268,63 @@ class Test_Model : public DNNTestLayer
             EXPECT_NEAR(box.angle, gtBox.angle, eps_angle);
         }
     }
+
+    void testYOLODetectionModel(const std::string& weights, const std::string& cfg, int yoloVersion,
+                                const std::string& imgPath, const std::vector<int>& refClassIds,
+                                const std::vector<float>& refConfidences,
+                                const std::vector<Rect2d>& refBoxes,
+                                double scoreDiff, double iouDiff,
+                                double confThreshold = 0.24, double nmsThreshold = 0.0,
+                                const Size& size = {-1, -1}, Scalar mean = Scalar(),
+                                double scale = 1.0, bool swapRB = false, bool crop = false,
+                                bool nmsAcrossClasses = false, float paddingValue = 0.0,
+                                ImagePaddingMode paddingMode = DNN_PMODE_NULL){
```

Comment:
Optional: You can pass `ImageToBlobParams` structure. It contains the same fields and has the reasonable defaults. Also the same could be added to the base class and your model, but let's postpone it till the next release.